### PR TITLE
Update framework (again) and remove some no-longer required workarounds

### PR DESCRIPTION
**Pull request by peppy**
Of note, iOS builds are failing here, but I want to PR to see what CI thinks.

### CODE REVIEW

**Comment by frenzibyte** [comment]
iOS build failures were caused by the combination of Veldrid and Realm, specifically due to the fact that Veldrid references a library that depends on an old version of `Microsoft.NetCore.PlatformAbstractions`, which, when combined with Realm, results in a downgrade package error. Manually choosing the latest version works and no problems occur on run time.

File: osu.Android.props [hunk 1]
```diff
@@ -10,11 +10,7 @@
     <EmbedAssembliesIntoApk>true</EmbedAssembliesIntoApk>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="ppy.osu.Framework.Android" Version="2022.1216.1" />
-  </ItemGroup>
-  <ItemGroup Label="Transitive Dependencies">
-    <!-- Realm needs to be directly referenced in all Xamarin projects, as it will not pull in its transitive dependencies otherwise. -->
-    <PackageReference Include="Realm" Version="10.18.0" />
+    <PackageReference Include="ppy.osu.Framework.Android" Version="2022.1219.0" />
   </ItemGroup>
```
Comment (frenzibyte) [inline osu.Android.props:14]: Has Android been checked with this removed? Slightly hesitant about it.

**Review by frenzibyte — APPROVED** [review]
Approving for iOS changes, will wait for second approval regarding Android.

File: osu.Android.props (hunk 1 above)
Comment (peppy) [reply osu.Android.props:14]: yep i tested on a device

File: osu.Android.props [hunk 2]
```diff
@@ -10,11 +10,7 @@
...
   <ItemGroup>
     <LinkDescription Include="$(MSBuildThisFileDirectory)\osu.Android\Linker.xml"/>
```
Comment (Susko3) [inline osu.Android.props:16]: This linker customisations on Android can also be removed. Tested on Release that this diff doesn't crash:
(would crash with xamarin)
```diff
diff --git a/osu.Game/Configuration/OsuConfigManager.cs b/osu.Game/Configuration/OsuConfigManager.cs
index 6cbb677a64..a6b587239b 100644
--- a/osu.Game/Configuration/OsuConfigManager.cs
+++ b/osu.Game/Configuration/OsuConfigManager.cs
@@ -5,6 +5,7 @@

 using System;
 using System.Diagnostics;
+using System.Globalization;
 using osu.Framework.Bindables;
 using osu.Framework.Configuration;
 using osu.Framework.Configuration.Tracking;
@@ -115,7 +116,7 @@ protected override void InitialiseDefaults()
             SetDefault(OsuSetting.MenuParallax, true);

             // See https://stackoverflow.com/a/63307411 for default sourcing.
-            SetDefault(OsuSetting.Prefer24HourTime, !CultureInfoHelper.SystemCulture.DateTimeFormat.ShortTimePattern.Contains(@"tt"));
+            SetDefault(OsuSetting.Prefer24HourTime, !new CultureInfo("th-TH").DateTimeFormat.ShortTimePattern.Contains(@"tt"));

             // Gameplay
             SetDefault(OsuSetting.PositionalHitsoundsLevel, 0.2f, 0, 1);
```

**Review by Susko3 — CHANGES_REQUESTED** [review]

File: osu.Android.props (hunk 2 above)
Comment (bdach) [reply osu.Android.props:16]: I say we do it separately (and also see if the iOS linker customisation can be removed too in the process, for full revert).

**Review by bdach — APPROVED** [review]
Tested Android to work

File: osu.Android.props (hunk 2 above)
Comment (frenzibyte) [reply osu.Android.props:16]: Yeah the iOS linker file shouldn't even exist, the file reference was removed in the project.

**Pull request merged**